torch/profiler/_utils.py

```diff
@@ -1,6 +1,9 @@
 from collections import deque
 from dataclasses import dataclass
 from typing import Dict, List
+import re
+
+import numpy as np
```

Comment:
While we are compatible with numpy, we don't like to use it as part of the internals because it we want to limit dependencies. (And lots of things fight over numpy version, so we don't want to be part of that brawl during install.) Everything that you're using numpy for you should also be able to just use `torch` functions.